cinch/bin/wrappers.py

```diff
@@ -44,19 +44,100 @@ def call_linchpin(work_dir, arg):
 
     :param work_dir: The linch-pin working directory that contains a PinFile
     and associated configuration files
-    :param args: An array of other command-line arguments to linchpin to pass
+    :param arg: A single argument to pass to the linchpin command
     :return: The exit code returned from linchpin, or 255 if errors come from
     elsewhere
     """
     # cinch will only support a subset of linchpin subcommands
-    supported_cmds = ['rise', 'drop']
+    supported_cmds = ['rise', 'drop', 'init']
     if arg not in supported_cmds:
         raise Exception('linchpin command "{0}" not '
                         'supported by cinch'.format(arg))
+
+    # If we are to ask linch-pin to interact with infrastructure we will check
+    # for some required configuration items and set up them for later use
+    if arg != 'init':
+        inventory_file = get_inventory(work_dir)
+        inventory_path = os.path.join(work_dir, 'inventory', inventory_file)
+
+    # For drop/teardown, we must run our teardown playbook(s) *before*
+    # linchpin terminates the instance(s)
+    if arg == 'drop':
+        call_ansible(inventory_path, 'teardown.yml')
+
+    # As of right now the linch-pin working directory is not configurable, so
+    # we essentially have a 'pushd' here to switch to the linch-pin working
+    # directory as needed. This code can be removed if the following issue is
+    # resolved:
+    # https://github.com/CentOS-PaaS-SIG/linch-pin/issues/119
+    @contextmanager
+    def pushd(new_dir):
+        previous_dir = os.getcwd()
+        os.chdir(new_dir)
+        yield
+        os.chdir(previous_dir)
+
+    # Execute the 'linchpin' command
+    with pushd(work_dir):
+        linchpin = local['linchpin']
+        exit_code = command_handler(linchpin, arg)
+
+    # Set up a linch-pin+cinch configuration skeleton for later use if the
+    # 'init' subcommand was executed previously
+    if arg == 'init':
+        # Consistent filename to use for various linch-pin YAML configurations
+        # for 'cinchpin'
+        config_file = 'cinch.yml'
+        # Cinch layout and topology paths to be added to linch-pin PinFile
+        config_setup = {
+            'cinch': {
+                'topology': config_file,
+                'layout': config_file
+            }
+        }
+        # These are the first-level config paths that linch-pin creates for us
+        # in the working directory
+        local_paths = ['layouts', 'topologies']
+        # Link to our docs with configuration examples
+        docs = 'http://example.com'
+        # Skeleton text to insert in YAML config files
+        skel_text = '''---
```

Comment:
Do we have a standard for putting multi-line strings outside of the flow (e.g. as file globals)? If not, I propose that we do so, as these lines interrupt the normal flow of Python code. We could even factor them out into a separate file if we end up with multiple such strings.